hack/verify-shellcheck.sh

```diff
@@ -36,13 +36,8 @@ SHELLCHECK_IMAGE="koalaman/shellcheck-alpine:v0.6.0@sha256:7d4d712a2686da99d3758
 all_shell_scripts=()
 while IFS=$'\n' read -r script;
   do git check-ignore -q "$script" || all_shell_scripts+=("$script");
-done < <(find . -name "*.sh" \
-  -not \( \
-    -path ./_\*      -o \
-    -path ./.git\*   -o \
-    -path ./vendor\* -o \
-    \( -path ./third_party\* -a -not -path ./third_party/forked\* \) \
-  \))
+done < <(grep -irl '#!.*sh' . \
+  --exclude-dir={/_\*,./.git\*,./vendor\*})
```

Comment:
confirmed this works with both BSD (macOS) and gnu grep

otherwise: (without the `.`)
```
grep: ./vendor/k8s.io/test-infra/prow/tide/README.md: No such file or directory
grep: ./vendor/k8s.io/test-infra/prow/crier/README.md: No such file or directory
grep: ./vendor/k8s.io/test-infra/prow/plank/README.md: No such file or directory
grep: ./vendor/k8s.io/test-infra/gubernator/github/secrets.py: No such file or directory
```

Reply:
ultimate nit / suggestion: this does fit on a single line, the `\` and second line are a little superfluous now :^)

Reply:
oh also should there be another `.` in front of`/_\*,`?
/hold

Reply:
Oops, not quite:
```
$ grep -irl '#!.*sh' . --exclude-dir={./_\*,./.git\*,./vendor\*}
./changelog-update
./push-build.sh
./hack/verify-shellcheck.sh
./.git/hooks/pre-receive.sample
./.git/hooks/pre-rebase.sample
./.git/hooks/pre-push.sample
./.git/hooks/update.sample
./.git/hooks/pre-commit.sample
./.git/hooks/commit-msg.sample
./.git/hooks/pre-applypatch.sample
./.git/hooks/prepare-commit-msg.sample
./.git/hooks/applypatch-msg.sample
./.git/hooks/post-update.sample
./rpm/entry.sh
./rpm/docker-build.sh
./script-template
./relnotes
./build/_test2/2
./gcbmgr
./_test/1
./debian/jenkins.sh
./debian/bionic/kubeadm/debian/postinst
./debian/bionic/kubectl/debian/postinst
./debian/bionic/kubelet/debian/postinst
./compile-release-tools
./branchff
./release-notify
./anago
./find_green_build
./lib/releaselib_test.sh
./lib/releaselib.sh
./lib/common.sh
./lib/gitlib_test.sh
./lib/gitlib.sh
./prin
```

I'm poking at it.

Reply:
You can see here, that it's checking the `.git` dir as well: https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_kind/670/pull-kind-verify/1144523142685790208

Reply:
yeah, I think this version is suitable for kind: `grep -irl '#!.*sh' . --exclude-dir={bin\*,.git\*,vendor\*}`, for here `grep -irl '#!.*sh' . --exclude-dir={_\*,.git\*,vendor\*}` maybe?

Reply:
Looks like we've got a winner: `grep -irl '#!.*sh' . --exclude-dir={_\*,.git\*,vendor\*}`